lib/cosmos/gui/line_graph/line_graph_drawing.rb

```diff
@@ -462,9 +466,19 @@ def update_graph_size
 
     def truncate_to_max(text, max_characters, value)
       if text.length > max_characters
+        # Transform the text into either decimal floating point or scientific
+        # notation, which ever results in a smaller string
         text = sprintf("%0.#{max_characters}g", value.to_f)
+        # If the value is still too big we need a smaller decimal
         if text.length > max_characters
-          text = sprintf("%0.#{max_characters - 5}g", value.to_f)
+          # Subtract 5 to get the correct number of characters
+          # For example: sprintf("%0.8g", 9123123123.123) => 9.1231231e+09
+          # We wanted an 8 character string but ended up with 13. We need to
+          # remove 5 of the decimal places so subtract 5.
+          #   sprintf("%0.3g", 9123123123.123) => 9.12e+09
```

Comment:
I figured out the -5 magic so I commented it. Also added code to handle a case of max_characters < 5 (would've crashed before)